api/client/samples/crop_models/brazil_soybeans.ipynb

```diff
@@ -0,0 +1,544 @@
+{
```

Comment:
<p>I think by running till this step, the dataframe `soy_ndvi_df` has duplicated rows (twice many rows as it should be), because the same NDVI series have been added by this cell and the above cell. Not sure how this will affect the following computations, but it doesnt hurt to be extra careful by calling `.drop_duplicates()`</p><p><br></p>

 _Reply via <a href='https://app.reviewnb.com/gro-intelligence/api-client/pull/58/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
<p>Not in this case, because this function only returns the weighted series it doesn't add it to the CropModel's dataframe. But yes, in the underlying GroClient, we dont check for duplicates, so if we add series more than once they will be duplicated. So here the only thing that is duplicated is province levle production, since we add it once when we call compute_weights (just to illustrate the weights), and one more time when we call compute_crop_weighted_series().</p><p><br></p><p>Also if you re-run  parts of the notebook (not from the beginning), it will duplicate some series.</p><p><br></p><p>The computations though are already robust to duplicates, and the results don't change. But the duplicates will make the frame larger. There's a separate PR coming whcih adds series indexing to the dataframe to improve that</p><p><br></p><p><br></p>

Reply:
https://github.com/gro-intelligence/api-client/pull/57